package.json

```diff
@@ -3,22 +3,25 @@
   "version": "1.0.1",
   "description": "Observe multiple React components with a single ResizeObserver.",
   "main": "dist/index.js",
+  "types": "dist/index.d.ts",
   "scripts": {
-    "build": "babel src/ -d dist/ --source-maps",
+    "src-type-definitions": "node scripts/src-type-definitions.js",
+    "dist-type-definitions": "node scripts/dist-type-definitions.js",
```

Comment:
I've not seen it done this way before, so I can't comment on this sorry.  

TS projects I've worked on have used the [Typescript compiler](https://www.typescriptlang.org/docs/handbook/compiler-options.html), which I see you're using in the `build` script here, but not to generate your type defs.

With `tsc`, you can just do this:
```
"scripts": {
    "build-type-definitions": "tsc --emitDeclarationOnly"
}
```

Is there a particular reason you have done it this way?

Reply:
I'm using `tsc` in the `build` script (it's wedged between some other commands).

These scripts are just for copying a dependency's type definitions for `ResizeObserver`, since TS has no concept of what a `ResizeObserver` is (or at least not one that matches the latest spec).

Reply:
Right, gotcha.  You're being explicit to work around TS being weird 👍   I'll try & take another look later, however I don't really know this package or Resize Observer well, so I'm thinking I'll try & test this via the breakpoints package, would that help?

Reply:
You'll need to do some `npm link` stuff that is a bit clonky sometimes. Test this in any other package locally to see if it informs your IDE of the types and if it will let you compile that package without blowing up :+1: